src/common/libutil/blobref.c

```diff
@@ -88,20 +89,22 @@ static void sha256_hash (const void *data, int data_len, void *hash, int hash_le
 
 /* true if s1 contains "s2-" prefix
  */
-static int prefixmatch (const char *s1, const char *s2)
+static bool prefixmatch (const char *s1, const char *s2)
 {
-    int len = strlen (s2);
-    if (strlen (s1) < len + 1 || s1[len] != '-')
-        return 0;
-    return !strncmp (s1, s2, len);
+    if (!strstarts (s1, s2))
+        return false;
+    s1 += strlen (s2);
+    if (*s1 != '-')
+        return false;
+    return true;
 }
```

Comment:
The old logic ensured that `s1[len]` would not be an array out of bounds, but I think in the new function the array out of bounds is possible.  Didn't look at context surrounding this function yet, just want to double check that array out of bounds isn't possible given how this is used.

Reply:
I'm not seeing how there could be an out of bounds access here?   If strlen(s1) < strlen(2) then strstarts() fails.  Thus strlen (s2) must be >= strlen (s1).  So s1 + strlen (s2) must be in bounds?

Reply:
hmmm, ok i think i was thinking of strlen(s1) == strlen(s2), so `strstarts()` passes.  but under that case I guess we just get the NUL delimeter, so all is good.